docs/user-guide/configuring/configuration-files.md

```diff
@@ -0,0 +1,394 @@
+
+The `rules` property can do any of the following to extend (or override) the set of rules:
+
+* enable additional rules
+* change an inherited rule's severity without changing its options:
+    * Base config: `"eqeqeq": ["error", "allow-null"]`
+    * Derived config: `"eqeqeq": "warn"`
+    * Resulting actual config: `"eqeqeq": ["warn", "allow-null"]`
+* override options for rules from base configurations:
+    * Base config: `"quotes": ["error", "single", "avoid-escape"]`
+    * Derived config: `"quotes": ["error", "single"]`
+    * Resulting actual config: `"quotes": ["error", "single"]
+
+### Using eslint-recommended
+
+An `extends` property value `"eslint:recommended"` enables a subset of core rules that report common problems, which have a checkmark (recommended) on the [rules page](https://eslint.org/docs/rules/). The recommended subset can change only at major versions of ESLint.
+
+If your configuration extends the recommended rules: after you upgrade to a newer major version of ESLint, review the reported problems before you use the `--fix` option on the [command line]((https://eslint.org/docs/user-guide/command-line-interface)#fix), so you know if a new fixable recommended rule will make changes to the code.
+
+The `eslint --init` command can create a configuration so you can extend the recommended rules.
+
+Example of a configuration file in JavaScript format:
+
+```js
+module.exports = {
+    "extends": "eslint:recommended",
+    "rules": {
+        // enable additional rules
+        "indent": ["error", 4],
+        "linebreak-style": ["error", "unix"],
+        "quotes": ["error", "double"],
+        "semi": ["error", "always"],
+
+        // override default options for rules from base configurations
+        "comma-dangle": ["error", "always"],
+        "no-cond-assign": ["error", "always"],
+
+        // disable rules from base configurations
+        "no-console": "off",
+    }
+}
+```
+
+### Using a shareable configuration package
+
+A [sharable configuration](https://eslint.org/docs/developer-guide/shareable-configs) is an npm package that exports a configuration object. Make sure the package has been installed to a directory where ESLint can require it.
+
+The `extends` property value can omit the `eslint-config-` prefix of the package name.
+
+The `eslint --init` command can create a configuration so you can extend a popular style guide (for example, `eslint-config-standard`).
+
+Example of a configuration file in YAML format:
+
+```yaml
+extends: standard
+rules:
+  comma-dangle:
+    - error
+    - always
+  no-empty: warn
+```
+
+### Using the configuration from a plugin
+
+A [plugin](https://eslint.org/docs/developer-guide/working-with-plugins) is an npm package that usually exports rules. Some plugins also export one or more named [configurations](https://eslint.org/docs/developer-guide/working-with-plugins#configs-in-plugins). Make sure the package has been installed in a directory where ESLint can require it.
```

Comment:
Plugins aren't just for rules and are intended to provide all kinds of extensions to ESLint. So rather than saying plugins normally export rules and only sometimes export configs, maybe we can say that plugins export a number of things including rules and configs?

Reply:
Oh yes. This'll def make it much better. Will the following content work?

> A [plugin](https://eslint.org/docs/developer-guide/working-with-plugins) is an npm package that can add various functions to ESLint. A plugin can perform numerous functions, including but not limited to the export of rules and [configurations](https://eslint.org/docs/developer-guide/working-with-plugins#configs-in-plugins). Make sure the package has been installed in a directory where ESLint can require it.